tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/ShapeOps.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.tensorflow.op.core;
+
+import java.util.Arrays;
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.op.Scope;
+import org.tensorflow.op.annotation.Endpoint;
+import org.tensorflow.op.annotation.Operator;
+import org.tensorflow.op.core.Concat;
+import org.tensorflow.op.core.ExpandDims;
+import org.tensorflow.op.core.Reshape;
+import org.tensorflow.op.core.Shape;
+import org.tensorflow.op.core.Size;
+import org.tensorflow.op.core.Slice;
+import org.tensorflow.op.math.FloorMod;
+import org.tensorflow.op.core.Constant;
+import org.tensorflow.op.core.Gather;
+import org.tensorflow.op.core.OnesLike;
+import org.tensorflow.op.core.ReduceProd;
+import org.tensorflow.op.core.Where;
+import org.tensorflow.op.dtypes.Cast;
+import org.tensorflow.op.math.NotEqual;
+import org.tensorflow.op.math.Sub;
+import org.tensorflow.types.TBool;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.types.family.TNumber;
+import org.tensorflow.types.family.TType;
+
+/**
+ * An operator working with org.tensorflow.op.core.Shape tensors
+ */
+@Operator
+public abstract class ShapeOps {
+
+    /**
+     * flatten the shape to 1 dimension
```

Comment:
For consistency with other classes, please start the doc summary with a capital letter.

Reply:
Ok